openedx/core/djangoapps/theming/models.py

```diff
@@ -17,3 +18,21 @@ class SiteTheme(models.Model):
 
     def __unicode__(self):
         return self.theme_dir_name
+
+    @staticmethod
+    def get_theme(site):
+        """
+        Get SiteTheme object for given site, returns default site theme if it can not
+        find a theme for the given site and `DEFAULT_SITE_THEME` setting has a proper value.
+
+        Args:
+            site (django.contrib.sites.models.Site): site object related to the current site.
+
+        Returns:
+            SiteTheme object for given site or a default site set by `DEFAULT_SITE_THEME`
+        """
+
+        theme = site.themes.first()
```

Comment:
The only problem I can see here is dealing with the cache timeout, which makes it hard to switch back and forth quickly between themes even using the Django admin -- so if we can work around (ie, ignore) that mechanism I think a URL parameter would be a useful add, yes -- if it ends up being a lot of work I think we'll need to log it as a follow-on story/feature, though.